src/components/ConditionalHeader/ConditionalHeader.js

```diff
@@ -1,19 +1,50 @@
 import { matchPath, useLocation } from 'react-router-dom'
 
+import { ArrowLeft } from 'react-feather'
 import BackButton from '../BackButton/BackButton'
 import Header from '../Header/Header'
+import { Link } from 'react-router-dom'
 import { currentSearchTermState } from '../../states'
-import { useRecoilValue } from 'recoil'
+import styled from 'styled-components'
+import { useRecoilState } from 'recoil'
 
 export default function ConditionalHeader() {
   const { pathname } = useLocation()
-  const currentSearchTerm = useRecoilValue(currentSearchTermState)
+  const [currentSearchTerm, setCurrentSearchTerm] = useRecoilState(
+    currentSearchTermState
+  )
 
   if (matchPath(pathname, { path: '/', exact: true })) {
     return <Header>PodcastFinder</Header>
   }
```

Comment:
You could this with a second router aswell, but I prefer your approach, just fyi